src/test/sigopcount_tests.cpp

```diff
@@ -28,6 +30,172 @@ Serialize(const CScript& s)
 
 BOOST_FIXTURE_TEST_SUITE(sigopcount_tests, BasicTestingSetup)
 
+// Feeds malformed PUSHDATA sequences to confirm the parser never crashes and still counts sig-ops that appear before the error.
+BOOST_AUTO_TEST_CASE(CountSigOpsErrors)
+{
+    for (const bool fAccurate : {false, true}) {
+        {
+            const auto script{CScript() << OP_CHECKSIG << opcodetype(0x4b)}; // push-75 with 0 bytes present
+            BOOST_CHECK_EQUAL(script.CountSigOps(fAccurate), 1);
```

Comment:
In commit "test: add `CountSigOps` edge-cases & known-template coverage" (c194618bb587a2e15ed11f268898599c74f1aa74)

The checks in this test are very repetitive. You might be able to make it easier understand and maintain by building a `std::vector<CScript>` of invalid scripts and looping over it.